mantidproject/mantid
# NexusDescriptor, LoadHelper :: Stop recursion on certain group entries

[files.zip](https://github.com/mantidproject/mantid/files/1011217/files.zip)


At ILL a new format of nexus files are planned to be produced for the scanning instruments.
The actual *scanned data* block is implemented using HDF5 hyper slabs, which is not in official nexus specification. 
When trying to read these blocks with regular nexus machinery one runs into a memory problem, that is tracked down to be inside the nexus code, and not in mantid or hdf (see the issue description).
This PR implements a protection relying on the custom NXclass names that are attributed to the new type of nexus files.
The aim is to stop the recursion in `NexusDescriptor::walkFile` ([L252](https://github.com/mantidproject/mantid/pull/19031/files#diff-5ac18ff3234bdf712052b6dc486ce826R252)) and `LoadHelper::recurseAndAddNexusFieldsToWsRun` ([L214](https://github.com/mantidproject/mantid/pull/19031/files#diff-cf91cb19ced37e956a8ebf60b08e65a7R214)) when hitting the custom entry for ILL data scan block.

The rest of the changes in this PR **do not** change any behaviour, they are some minor improvements for readability and error handling.





**To test:**


1. Merge in branch #18868 and build.
2. Set the facility to ILL.
3. Try to load all the attached files with the generic `Load`.
4. There should be no crash, neither after closing MantidPlot.

Fixes #19009 .

Does not need to be in the release notes.



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
